pkg/s3select/json/reader.go

```diff
@@ -73,8 +73,9 @@ type objectReader struct {
 	endOfObject bool
 }
 
-func (or *objectReader) objectEndIndex(p []byte, length int) int {
-	for i := 0; i < length; i++ {
+// Returns the location of the end curly braces in a given buffer.
+func (or *objectReader) findEndIndex(p []byte) int {
```

Comment:
The idea of the function name is from `strings.Index()`.  If `objectEndIndex()` is confusing, you could have `jsonEndIndex()` or simply `endIndex()` i.e. no need to have `find` and `index` together.